unit_tests/src/test_atomic_apply_metadata_operations.cpp

```diff
@@ -0,0 +1,478 @@
+                    {"value", "the_val"}
+                },
+                {
+                    {"operation", "remove"},
+                    {"attribute", "name"},
+                    {"value", "john doe"}
+                },
+                {
+                    {"operation", "remove"},
+                    {"attribute", "name"},
+                    {"value", "john doe"}
+                },
+                {
+                    {"operation", "remove"},
+                    {"attribute", "occupation"},
+                    {"value", "software developer"}
+                }
+            })}
+        }.dump();
+
+        char* json_error_string{};
+
+        REQUIRE(rc_atomic_apply_metadata_operations(&_conn, json_input.c_str(), &json_error_string) == 0);
+        REQUIRE(json_error_string == "{}"s);
+    }
+
+    {
+        const auto json_input = json{
+            {"entity_name", _env.rodsUserName},
+            {"entity_type", "user"},
+            {"operations", json::array({
+                {
+                    {"operation", "remove"},
+                    {"attribute", "irods_user"},
+                    {"value", _env.rodsUserName},
+                    {"units", "rodsadmin"}
+                },
+                {
+                    {"operation", "remove"},
+                    {"attribute", "renci_position"},
+                    {"value", "research_developer"}
+                },
+                {
+                    {"operation", "remove"},
+                    {"attribute", "name_does_not_exist"},
+                    {"value", "value_does_not_exist"},
+                    {"units", "units_does_not_exist"}
+                }
+            })}
+        }.dump();
+
+        char* json_error_string{};
+
+        REQUIRE(rc_atomic_apply_metadata_operations(&_conn, json_input.c_str(), &json_error_string) == 0);
+        REQUIRE(json_error_string == "{}"s);
+    }
+
+    {
+        const auto json_input = json{
+            {"entity_name", _env.rodsDefResource},
+            {"entity_type", "resource"},
+            {"operations", json::array({
+                {
+                    {"operation", "remove"},
+                    {"attribute", "this_was_set_atomically"},
+                    {"value", "it worked!"},
+                },
+                {
+                    {"operation", "remove"},
+                    {"attribute", "the_atomic_attribute_name"},
+                    {"value", "the_atomic_value"},
+                    {"units", "the_atomic_units"}
+                },
+                {
+                    {"operation", "remove"},
+                    {"attribute", "this_was_set_atomically"},
+                    {"value", "it worked!"}
+                }
+            })}
+        }.dump();
+
+        char* json_error_string{};
+
+        REQUIRE(rc_atomic_apply_metadata_operations(&_conn, json_input.c_str(), &json_error_string) == 0);
+        REQUIRE(json_error_string == "{}"s);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [String literal compared with variable 'json_error_string'. Did you intend to use strcmp() instead?](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5179166)